avocado/core/runner.py

```diff
@@ -332,6 +332,18 @@ def sigterm_handler(signum, frame):     # pylint: disable=W0613
         self.job._result_events_dispatcher.map_method('start_test',
                                                       self.result,
                                                       early_state)
+        if getattr(self.job.args, 'list_test_data_directories', False):
+            data_sources = getattr(instance, "DATA_SOURCES", [])
+            if data_sources:
+                locations = []
+                for source in data_sources:
+                    locations.append(instance.get_data("", source=source,
+                                                       must_exist=False))
```

Comment:
You can still keep this if `DATA_SOURCES` (btw incorrect name) stored names+paths together. (I know I suggested `_data_dirs` but `data_dirs` would be probably necessary to allow implementation like this.